kafka-ui-react-app/src/components/Topics/Topic/Details/Messages/Filters/EditFilter.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { InputLabel } from 'components/common/Input/InputLabel.styled';
+import Input from 'components/common/Input/Input';
+import { Textarea } from 'components/common/Textbox/Textarea.styled';
+import { useForm, FormProvider, Controller } from 'react-hook-form';
+import { ErrorMessage } from '@hookform/error-message';
+import { Button } from 'components/common/Button/Button';
+import yup from 'lib/yupExtended';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as S from 'components/Topics/Topic/Details/Messages/Filters/Filters.styled';
+import { MessageFilters } from 'components/Topics/Topic/Details/Messages/Filters/Filters';
+import { FilterEdit } from 'components/Topics/Topic/Details/Messages/Filters/FilterModal';
+
+const validationSchema = yup.object().shape({
+  name: yup.string().required(),
+  code: yup.string().required(),
+});
+
+export interface EditFilterProps {
+  editFilter: FilterEdit;
+  toggleEditModal(): void;
+  editSavedFilter(filter: FilterEdit): void;
+}
+
+const EditFilter: React.FC<EditFilterProps> = ({
+  editFilter,
+  toggleEditModal,
+  editSavedFilter,
+}) => {
+  const methods = useForm<MessageFilters>({
+    mode: 'onChange',
+    resolver: yupResolver(validationSchema),
+  });
+  const {
+    handleSubmit,
+    control,
+    formState: { isDirty, isSubmitting, isValid, errors },
+  } = methods;
+  const onSubmit = React.useCallback(
+    async (values: MessageFilters) => {
+      editSavedFilter({ index: editFilter.index, filter: values });
+      toggleEditModal();
+    },
+    [editSavedFilter, editFilter.index, toggleEditModal]
+  );
+  return (
+    <>
+      <S.FilterTitle>Edit saved filter</S.FilterTitle>
```

Comment:
This part is very Similar to the second part in the AddFilter maybe we can refactor and write an independent test to it.